File: 3484-lexicographically-smallest-string-after-a-swap/lexicographically-smallest-string-after-a-swap.py
class Solution:
    def getSmallestString(self, s: str) -> str:
        ans = [i for i in s]
        n = len(ans)

        for i in range(n - 1):
            x = int(ans[i])
            y = int(ans[i + 1])

            if x & 1 == y & 1 and x > y:
                ans[i + 1] = str(x)
                ans[i] = str(y)
                break

        return "".join(ans)

File: 3484-lexicographically-smallest-string-after-a-swap/lexicographically-smallest-string-after-a-swap.js
/**
 * @param {string} s
 * @return {string}
 */
var getSmallestString = function(s) {
    var arr = [];

    for(let i = 0; i < s.length; i++){
        arr.push(Number(s[i]));
    }

    let n = arr.length;

    for(let i = 0; i < n - 1; i++){
        let x = arr[i];
        let y = arr[i + 1];

        if((x & 1) === (y & 1) && x > y){
            arr[i] = y;
            arr[i + 1] = x;
            break;
        }
    }

    return arr.join("");
};